### PR TITLE
Add expense command

### LINKED ISSUE #60: Add Expense Command
Add a command to add an expense to FastTrack

### PR DESCRIPTION
**Pull request by nicleejy**
Added the following features
1. Add expense
2. Delete expense 
3. Add category
4. Delete category
Other changes include UI overhaul, storage modifications and addition of new methods and fields inside Model, Logic and AddressBook
Closes #60 #61 #65 #56 #51 #50 #5 #6 #2 #4 #47

### CODE REVIEW

**Comment by nicleejy** [comment]
Note: To write test cases and improve upon JavaDoc once any unused classes have been removed and code cleanup

File: .gitignore
Comment (shirsho-12) [inline .gitignore:1]: Add addressbook.json to gitignore.

**Review by shirsho-12 — APPROVED** [review]
Future PRs should not be this long. Too many files to review in one go.

File: .gitignore
Comment (gitsac) [reply .gitignore:1]: LGTM, standard data to be accessed should be standardized.

File: src/main/java/seedu/address/logic/commands/AddCategory.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * Adds a person to the address book.
+ */
+public class AddCategory extends Command {
+
+    public static final String COMMAND_WORD = "addcat";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a category to FastTrack. "
+            + "Parameters: "
+            + PREFIX_CATEGORY + "CATEGORY_NAME"
```
Comment (gitsac) [inline src/main/java/seedu/address/logic/commands/AddCategory.java:20]: LGTM, good use of final string to show how the input formatting should be.

File: src/main/java/seedu/address/model/category/UniqueCategoryList.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    public ObservableList<Category> asUnmodifiableList() {
+        return this.internalUnmodifiableList;
+    }
```
Comment (gitsac) [inline src/main/java/seedu/address/model/category/UniqueCategoryList.java:91]: JavaDocs lacking @params and @returns, TODO in future pull-request.

**Review by gitsac — APPROVED** [review]

**Pull request merged**